Error parsing French accents

Tested build: http://fgpv.cloudapp.net/demo/develop/prod/samples/index-mobile.html

Opening a datatable for any layer which has columns with French accents in them throws an error: 
`Error: [$parse:lexerr] Lexer Error: Unexpected next character  at columns 8-8 [é] in expression [self.Catégorie.value].`

More errors of the same type when adding http://fgpv.cloudapp.net/demo/__assets__/Sample%20Data/G%C3%A9ochronologie.csv.

> rv-main.js:95762 Error: [$parse:lexerr] Lexer Error: Unterminated quote at columns 11-29 ['échantillon.value] in expression [self.Num. d'échantillon.value].
http://errors.angularjs.org/1.6.4/$parse/lexerr?p0=Unterminated%20quote&p1=…29%20%5B'%C3%A9chantillon.value%5D&p2=self.Num.%20d'%C3%A9chantillon.value
    at rv-main.js:81303

--- a/src/app/ui/table/table-definition.directive.js
+++ b/src/app/ui/table/table-definition.directive.js
@@ -270,8 +270,11 @@ function rvTableDefinition(stateManager, events, $compile, tableService, layoutS
             filterScope.self = filter.self;
             filter.scope = filterScope;
 
+            // Using default column name results in errors with specific characters (accents, brackets, etc.) for template data binding
+            // Need to provide a simplified column name for data binding and also keep a reference to the default name
+            // https://github.com/fgpv-vpgf/fgpv-vpgf/issues/2019
             if (!column.simpleColumnName) {
-                const simpleColumnName = 'a' + counter;     // set a simplified column name for parsing purposes
+                const simpleColumnName = 'a' + counter;
                 column.simpleColumnName = simpleColumnName;
                 columnNameMap[simpleColumnName] = column.name;
                 counter++;

--- a/src/app/ui/table/table-definition.directive.js
+++ b/src/app/ui/table/table-definition.directive.js
@@ -53,7 +53,8 @@ const FILTERS = {
     }
 };
 
-const map = { };
+const columnNameMap = { };
+let counter = 0;
 
 // jscs:disable maximumLineLength
 const FILTERS_TEMPLATE = {
@@ -62,7 +63,7 @@ const FILTERS_TEMPLATE = {
             <md-input-container class="md-block" md-no-float flex>
                 <input ng-click="self.prevent($event)"
                         ng-keypress="self.prevent($event)"
-                        ng-change="self.change('${map[column]}', self.${column}.value)"
+                        ng-change="self.change('${columnNameMap[column]}', self.${column}.value)"
                         ng-model="self.${column}.value" class="ng-pristine ng-valid md-input ng-touched" placeholder="{{ self.placeholder | translate }}"
                         ng-disabled="self.${column}.static" />
             </md-input-container>
@@ -72,7 +73,7 @@ const FILTERS_TEMPLATE = {
             <md-input-container class="md-block" md-no-float flex>
                 <md-select ng-click="self.prevent($event)"
                     ng-model="self.${column}.value"
-                    md-on-close="self.change('${map[column]}', self.${column}.value)"
+                    md-on-close="self.change('${columnNameMap[column]}', self.${column}.value)"
                     ng-disabled="self.${column}.static"
                     placeholder="{{ self.placeholder | translate }}" multiple>
                     <md-option ng-repeat="value in self.${column}.values" ng-value="value">
@@ -86,14 +87,14 @@ const FILTERS_TEMPLATE = {
             <md-input-container class="md-block" md-no-float flex>
                 <input rv-table-number-only
                         ng-click="self.prevent($event)"
-                        ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
+                        ng-change="self.change('${columnNameMap[column]}', self.${column}.min, self.${column}.max)"
                         ng-model="self.${column}.min" class="ng-pristine ng-valid md-input ng-touched" placeholder="{{ self.min.placeholder | translate }}"
                         ng-disabled="self.${column}.static" />
             </md-input-container>
             <md-input-container class="md-block" md-no-float flex>
                 <input rv-table-number-only
                         ng-click="self.prevent($event)"
-                        ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
+                        ng-change="self.change('${columnNameMap[column]}', self.${column}.min, self.${column}.max)"
                         ng-model="self.${column}.max" class="ng-pristine ng-valid md-input ng-touched" placeholder="{{ self.max.placeholder | translate }}"
                         ng-disabled="self.${column}.static" />
             </md-input-container>
@@ -102,14 +103,14 @@ const FILTERS_TEMPLATE = {
         `<div class="rv-filter-date" ng-show="self.${column}.filtersVisible">
             <md-datepicker
                 ng-click="self.prevent($event)"
-                ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
+                ng-change="self.change('${columnNameMap[column]}', self.${column}.min, self.${column}.max)"
                 ng-model="self.${column}.min"
                 md-placeholder="{{ self.min.placeholder | translate }}"
                 ng-disabled="self.${column}.static">
             </md-datepicker>
             <md-datepicker
                 ng-click="self.prevent($event)"
-                ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
+                ng-change="self.change('${columnNameMap[column]}', self.${column}.min, self.${column}.max)"
                 ng-model="self.${column}.max"
                 md-placeholder="{{ self.max.placeholder | translate }}"
                 ng-disabled="self.${column}.static">
@@ -270,15 +271,10 @@ function rvTableDefinition(stateManager, events, $compile, tableService, layoutS
             filter.scope = filterScope;
 
             if (!column.simpleColumnName) {
-                if (typeof window.counter !== 'undefined') {
-                    window.counter++;
-                } else {
-                    window.counter = 0;
-                }
-
-                const simpleColumnName = 'a' + window.counter;
-                column.simpleColumnName = simpleColumnName
-                map[simpleColumnName] = column.name;
+                const simpleColumnName = 'a' + counter;     // set a simplified column name for parsing purposes
+                column.simpleColumnName = simpleColumnName;
+                columnNameMap[simpleColumnName] = column.name;
+                counter++;
             }
 
             filter.scope.self[column.simpleColumnName] = column.filter;

--- a/src/app/ui/table/table-definition.directive.js
+++ b/src/app/ui/table/table-definition.directive.js
@@ -53,6 +53,8 @@ const FILTERS = {
     }
 };
 
+const map = { };
+
 // jscs:disable maximumLineLength
 const FILTERS_TEMPLATE = {
     string: column =>
@@ -60,7 +62,7 @@ const FILTERS_TEMPLATE = {
             <md-input-container class="md-block" md-no-float flex>
                 <input ng-click="self.prevent($event)"
                         ng-keypress="self.prevent($event)"
-                        ng-change="self.change('${column}', self.${column}.value)"
+                        ng-change="self.change('${map[column]}', self.${column}.value)"
                         ng-model="self.${column}.value" class="ng-pristine ng-valid md-input ng-touched" placeholder="{{ self.placeholder | translate }}"
                         ng-disabled="self.${column}.static" />
             </md-input-container>
@@ -70,7 +72,7 @@ const FILTERS_TEMPLATE = {
             <md-input-container class="md-block" md-no-float flex>
                 <md-select ng-click="self.prevent($event)"
                     ng-model="self.${column}.value"
-                    md-on-close="self.change('${column}', self.${column}.value)"
+                    md-on-close="self.change('${map[column]}', self.${column}.value)"
                     ng-disabled="self.${column}.static"
                     placeholder="{{ self.placeholder | translate }}" multiple>
                     <md-option ng-repeat="value in self.${column}.values" ng-value="value">
@@ -84,14 +86,14 @@ const FILTERS_TEMPLATE = {
             <md-input-container class="md-block" md-no-float flex>
                 <input rv-table-number-only
                         ng-click="self.prevent($event)"
-                        ng-change="self.change('${column}', self.${column}.min, self.${column}.max)"
+                        ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
                         ng-model="self.${column}.min" class="ng-pristine ng-valid md-input ng-touched" placeholder="{{ self.min.placeholder | translate }}"
                         ng-disabled="self.${column}.static" />
             </md-input-container>
             <md-input-container class="md-block" md-no-float flex>
                 <input rv-table-number-only
                         ng-click="self.prevent($event)"
-                        ng-change="self.change('${column}', self.${column}.min, self.${column}.max)"
+                        ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
                         ng-model="self.${column}.max" class="ng-pristine ng-valid md-input ng-touched" placeholder="{{ self.max.placeholder | translate }}"
                         ng-disabled="self.${column}.static" />
             </md-input-container>
@@ -100,14 +102,14 @@ const FILTERS_TEMPLATE = {
         `<div class="rv-filter-date" ng-show="self.${column}.filtersVisible">
             <md-datepicker
                 ng-click="self.prevent($event)"
-                ng-change="self.change('${column}', self.${column}.min, self.${column}.max)"
+                ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
                 ng-model="self.${column}.min"
                 md-placeholder="{{ self.min.placeholder | translate }}"
                 ng-disabled="self.${column}.static">
             </md-datepicker>
             <md-datepicker
                 ng-click="self.prevent($event)"
-                ng-change="self.change('${column}', self.${column}.min, self.${column}.max)"
+                ng-change="self.change('${map[column]}', self.${column}.min, self.${column}.max)"
                 ng-model="self.${column}.max"
                 md-placeholder="{{ self.max.placeholder | translate }}"
                 ng-disabled="self.${column}.static">
@@ -266,16 +268,29 @@ function rvTableDefinition(stateManager, events, $compile, tableService, layoutS
             const filterScope = scope.$new(true);
             filterScope.self = filter.self;
             filter.scope = filterScope;
-            filter.scope.self[column.name] = column.filter;
 
-            $rootScope.$watch(() => layoutService.isFiltersVisible, val => { filter.self[column.name].filtersVisible = val });
+            if (!column.simpleColumnName) {
+                if (typeof window.counter !== 'undefined') {
+                    window.counter++;
+                } else {
+                    window.counter = 0;
+                }
+
+                const simpleColumnName = 'a' + window.counter;
+                column.simpleColumnName = simpleColumnName
+                map[simpleColumnName] = column.name;
+            }
+
+            filter.scope.self[column.simpleColumnName] = column.filter;
+
+            $rootScope.$watch(() => layoutService.isFiltersVisible, val => { filter.self[column.simpleColumnName].filtersVisible = val });
 
             // create directive
-            const template = FILTERS_TEMPLATE[column.type](column.name);
+            const template = FILTERS_TEMPLATE[column.type](column.simpleColumnName);
 
             // return directive, scope and column type
             return { directive: $compile(template)(filter.scope),
-                    scope: filter.scope.self[column.name] };
+                    scope: filter.scope.self[column.simpleColumnName] };
         }
 
         /**

